transport/grpc/integration_test.go

```diff
@@ -216,8 +296,8 @@ func TestApplicationErrorPropagation(t *testing.T) {
 	})
 }
 
-func doWithTestEnv(t *testing.T, transportOptions []TransportOption, inboundOptions []InboundOption, outboundOptions []OutboundOption, f func(*testing.T, *testEnv)) {
-	testEnv, err := newTestEnv(transportOptions, inboundOptions, outboundOptions)
+func doWithTestEnv(t *testing.T, transportOptions []TransportOption, inboundOptions []InboundOption, outboundOptions []OutboundOption, dialOptions []DialOption, f func(*testing.T, *testEnv)) {
```

Comment:
not actionable for this PR: This code can probably be simplified in the future
by using the unifying grpc.Option type instead of the 4 different option
groups.

```
func newTestEnv(opts ...Option) (*testEnv, error) {
      for _, opt := range opts {
        switch o := opt.(type) {
        case TransportOption:
            ...
        case InboundOption:
            ...
        ...
        }
    }
    ...
```